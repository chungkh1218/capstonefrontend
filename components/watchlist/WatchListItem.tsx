/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import * as React from "react";
import { Component } from "react";
// import React, { Component } from "react";
import {
  Platform,
  StyleSheet,
  Text,
  View,
  Dimensions,
  Image
} from "react-native";

const instructions = Platform.select({
  ios: "Press Cmd+R to reload,\n" + "Cmd+D or shake for dev menu",
  android:
    "Double tap R on your keyboard to reload,\n" +
    "Shake or press menu button for dev menu"
});

interface IWatchListItemProps {
  re_id: number;
  catfathername: string;
  catname: string;
  my_target_price: string;
  addr: string;
  area: string;
}
export default class WatchListItem extends Component<IWatchListItemProps> {
  constructor(props: IWatchListItemProps) {
    super(props);
  }
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.imagePanel}>
          <Image
            source={{
              uri: "https://reactjs.org/logo-og.png"
            }}
            style={{ width: 256, height: 256 }}
          />
        </View>
        <View style={styles.userPanel}>
          <Text style={styles.instructions}>{this.props.re_id}</Text>
          <Text style={styles.instructions}>{this.props.catname}</Text>
          <Text style={styles.instructions}>{this.props.catfathername}</Text>
          <Text style={styles.instructions}>{this.props.area}</Text>
          <Text style={styles.instructions}>{this.props.addr}</Text>
          <Text style={styles.instructions}>{this.props.my_target_price}</Text>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#F5FCFF"
  },
  welcome: {
    fontSize: 20,
    textAlign: "center",
    margin: 10
  },
  instructions: {
    textAlign: "center",
    color: "#333333",
    marginBottom: 5
  },
  imagePanel: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    width: Dimensions.get("window").width,
    height: 100,
    backgroundColor: "red"
  },
  userPanel: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    width: Dimensions.get("window").width,
    height: 100,
    backgroundColor: "beige"
  }
});
