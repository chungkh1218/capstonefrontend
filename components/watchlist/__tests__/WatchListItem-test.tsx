import React from "react";
import WatchListItem from "../WatchListItem";

import renderer from "react-test-renderer";

test("WatchListItem renders correctly", () => {
  const tree = renderer
    .create(<WatchListItem catname catfathername avWinloss avPrice_sq />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
