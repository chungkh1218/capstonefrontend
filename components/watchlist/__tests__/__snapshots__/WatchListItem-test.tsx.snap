// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchListItem renders correctly 1`] = `
<View>
  <Text
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "letterSpacing": 1,
        },
      ]
    }
  />
  <View
    style={
      Object {
        "color": "dimgray",
        "fontSize": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    />
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    />
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Winloss: 
      %
    </Text>
    <Text
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Price: $
    </Text>
  </View>
</View>
`;
